crates/networking/discv5/src/subnet.rs

```diff
@@ -137,6 +143,49 @@ impl Decodable for SyncCommitteeSubnets {
     }
 }
 
+/// Compute a single subscribed subnet based on node_id, epoch, and index
+pub fn compute_subscribed_subnet(node_id: NodeId, epoch: u64, index: usize) -> u8 {
+    // Extract prefix from first 8 bytes of node_id
+    let node_id_prefix = u64::from_be_bytes(node_id.raw()[..8].try_into().unwrap())
+        >> (64 - ATTESTATION_SUBNET_PREFIX_BITS);
+    let node_offset = u64::from_be_bytes(node_id.raw()[24..32].try_into().unwrap())
+        % EPOCHS_PER_SUBNET_SUBSCRIPTION;
+    let permutation_seed =
+        Sha256::digest(((epoch + node_offset) / EPOCHS_PER_SUBNET_SUBSCRIPTION).to_le_bytes());
+    let permutated_prefix = compute_shuffled_index(
+        node_id_prefix as usize,
+        1 << ATTESTATION_SUBNET_PREFIX_BITS,
+        permutation_seed.into(),
+    );
+    ((permutated_prefix + index) % ATTESTATION_SUBNET_COUNT) as u8
+}
+
+/// Compute all subscribed subnets for a node
+pub fn compute_subscribed_subnets(node_id: NodeId, epoch: u64) -> Vec<u8> {
+    (0..SUBNETS_PER_NODE)
+        .map(|i| compute_subscribed_subnet(node_id, epoch, i))
+        .collect()
+}
+
+/// Simple shuffling function based on Ethereum's shuffle algorithm
+fn compute_shuffled_index(index: usize, index_count: usize, seed: [u8; 32]) -> usize {
+    assert!(index < index_count, "Index out of bounds");
+    let mut indices = (0..index_count).collect::<Vec<_>>();
+    let mut i = 0;
+    while i < index_count {
+        let pivot = {
+            let mut hasher = Sha256::new();
+            hasher.update(seed);
+            hasher.update((i as u32).to_le_bytes());
+            let hash = hasher.finalize();
+            u64::from_le_bytes(hash[..8].try_into().unwrap()) as usize % (index_count - i)
+        };
+        indices.swap(i, i + pivot);
+        i += 1;
+    }
+    indices[index]
+}
```

Comment:
This function has already been implemented please remove this compute_shuffled_index